simon.sh

```diff
@@ -273,6 +276,62 @@ not_t1a0() {
     [ 1 = "$FSMPL" ] && [ 0 = "$FAUTO" ] && return 1 || return 0
 }
 
+##
+# The function has to be executed in background to read user input from
+# $2. The input is stores in file $1.
+#
+# $1 - filename to store the input (TODO use proper IPC, such as signals)
+# $2 - terminal device (e.g., /dev/tty1, /dev/pts/3 and so on)
+read_from_term()
+{
+    printf '_skip_' >"$tmp_file"
+    read -r yn <"$2"
+    printf -- '%s' "${yn}" >"$1"
+}
+
+##
+# Executes read_from_term() in background, goes to sleep and polls user
+# input from a file. The function modifies TMP variable to "return" the
+# result.
+get_answer()
+{
+    local tmp_file rc dev_tty
+    tmp_file=/tmp/simon_answer$(date +%m%d%H%M)
+    dev_tty=$(tty)
+    FPUTC=1
+
+    read_from_term "$tmp_file" "$dev_tty" &
+    while true; do
+        sleep 0.1
+        yn=$(cat "$tmp_file" 2>/dev/null)
```

Comment:
wouldn't it be better to check if `tmp_file` exists and then `cat` it. No need to `stderr` redirection.

Even better: there would be a bug if the flie is not created (since `read_from_term` was not scheduled on execution) and `yn` would be empty -> break from the loop.

Reply:
see 8284f90